frontend/src/ts/controllers/sound-controller.ts

```diff
@@ -341,6 +347,71 @@ function initAudioContext(): void {
   }
 }
 
+const scales: Record<string, string[]> = {
```

Comment:
Then you can do:

```suggestion
const scales: Record<string, ValidNotes[]> = {
  pentatonic: ["C", "D", "E", "G", "A"],
  wholetone: ["C", "D", "E", "Gb", "Ab", "Bb"]
};
```

This gives you nice types when defining a new scale :)

Reply:
That is actually brilliant. Nice.